### PR TITLE
#5382: Delete deprecated overloads of Kokkos::sort() taking parameter 'bool always_use_kokkos_sort'

### PR DESCRIPTION
**Pull request by PhilMiller**
A richer sorting interface will be implemented later, probably involving a handle class of some sort. This is just clearing out an obsolete design.

### CODE REVIEW

File: algorithms/src/Kokkos_Sort.hpp [hunk 1]
```diff
@@ -628,12 +624,38 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
   bin_sort.sort(exec, view);
 }
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
+template <class ExecutionSpace, class ViewType>
+KOKKOS_DEPRECATED_WITH_COMMENT(
```
Comment (PhilMiller) [inline algorithms/src/Kokkos_Sort.hpp:629]: Since `develop` is now on track for 4.0, I guess I should drop the deprecated code entirely?

**Comment by PhilMiller** [comment]
I added a commit to just delete the deprecated code. I can drop that commit if we decide to keep it for longer (either 4.1, or 5.0)

**Comment by masterleinad** [comment]
https://github.com/kokkos/kokkos/pull/5372 pretty much does this already. I'm not sure if we need a separate pull request.

**Comment by masterleinad** [comment]
Otherwise, this should also remove `Impl::try_std_sort` (like #5372 does).

**Comment by PhilMiller** [comment]
Until that's ready to merge, this ensures that `develop` is consistent with what we do on the release branch with #5384

**Review by masterleinad — APPROVED** [review]

**Comment by fnrizzi** [comment]
what is going on with the CI ?

**Review by fnrizzi — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**